### ISSUE TITLE
getState not called (and state not updated) after plugin changes it's own state

### ISSUE THREAD

**Issue by nicwise** [nfarina/homebridge#1355]
Hi!

I've been writing a homebridge platform for Amazon SQS (which allows me to hook up a Dash button to AWS IOT -> Lambda -> SQS -> my code -> homebridge, or anything else that can post to SQS).

The SQS bit is easy, but I have an issue setting the state in the Home app.

My platform code is here, and quite simple: https://github.com/nicwise/homebridge-platform-sqs

So if I press the button, it fires the plugin code:

```
[2017-6-24 18:24:53] [DashSQS] Path Button: SQS Button state change. New state is false
[2017-6-24 18:24:53] [DashSQS] Path Button: Set State to false
```
Note that it's not calling getState :(

However, the Home app doesn't show the button state change until I swipe up Notification Center, or close and re-open the Home app. Ditto any Automation's don't fire.

The rest of my setup is: iOS 10.3.2, with the current AppleTV OS acting as the host for the house.

ALSO, here's the kicker:

I develop off my laptop, so I have an install of homebridge on there (the one which isn't working on my a Mac Mini), and I have both bridges installed into the Home app. But the one on my laptop _works_. the other one doesn't. Both, however, are the same code - I do this to install it on the mini

```
git clone https://github.com/nicwise/homebridge-platform-sqs.git
npm install -g homebridge-platform-sqs
```

Any ideas? Am I missing something in the platform creation? (UUID maybe?)

**Comment by nicwise** [comment]
Tried using the Elgato Eve app as the front end, and same same - I have to force a refresh for it to show up. Not sure if it's HomeKit, Homebridge, the AppleTV.... or something else.

SO SO frustrating that it's working on one machine, and not on the other. (and that the Automation's don't work - they did a couple of days ago!)

**Comment by snowdd1** [comment]
Hi @nicwise ,
External events like a push button are supposed to call setValue or updateValue in the plugin code. If your Homekit app has itself correctly registered to receive event notifications, then it's supposed to show these immediately.  
The get...() functions are called when the Homekit app *actively* asks for the status (because app is activated from being backgrounded, or opened, or you've asked Siri for the state of something etc. So it is **not** called when you just push your button.

**Comment by nicwise** [comment]
Thanks @snowdd1 - thats good to know. Looks like my issue is node version. More at #1353 

```
OK, I've found the/my problem. I don't have a fix tho. At least, not a good one.

It's Node. 8.1.2 to be exact. Well, thats the version I used to have.

TL;DR: 
* It works in 6.11, but my `homebridge-broadlink-rm` doesn't. That needs 7.10
* It works in 7.10 !! :)
* It FAILS in 8.1.2

I run the `homebridge-broadlink-rm` plugin which need 7.6.0 or better, so I got the latest stable, which appears to show this issue

I went back to 6.10.3, and everything works (I had to disable the broadlink plugin tho :( )

My other machine, which is working, is running 4.6.0. I need to upgrade that to 6.10 :)

I then went for the latest in 7.x (for the broadlink one) which WORKS! yay!

So.... 8.1.2 bug? I'm not overly familiar with the Node ecosystem, so I don't know where to look for breaking changes....
```
